torch/autograd/_functions/reduce.py

```diff
@@ -46,7 +46,7 @@ def forward(self, input):
     def backward(self, grad_output):
         if self.dim is None:
             input, = self.saved_tensors
-            grad_input = grad_output.new(self.input_size).fill_(self.result)
+            grad_input = grad_output.new(self.input_size).fill_(self.result * grad_output[0])
```

Comment:
This leads to a sync point in the GPU because of the `grad_output[0]`, right?
A solution that doesn't have a sync point would be something like
```python
grad_input = grad_output.expand(self.input_size).mul(self.result)
```

But now I see that other modules (like Sum) also access `grad_output[0]`. Maybe we should change that?